bugsnag-android-performance/src/test/java/com/bugsnag/android/performance/internal/AutoInstrumentationCacheTest.kt

```diff
@@ -0,0 +1,138 @@
+package com.bugsnag.android.performance.internal
+
+import android.app.Activity
+import android.app.Application
+import android.content.Context
+import android.content.pm.PackageInfo
+import android.content.pm.PackageManager
+import com.bugsnag.android.performance.AutoInstrumentationCache
+import com.bugsnag.android.performance.DoNotEndAppStart
+import com.bugsnag.android.performance.DoNotInstrument
+import com.bugsnag.android.performance.PerformanceConfiguration
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.junit.Test
+import org.mockito.kotlin.any
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.eq
+import org.mockito.kotlin.mock
+
+internal class AutoInstrumentationCacheTest {
+    private val autoInstrumentationCache = AutoInstrumentationCache()
+
+    @DoNotInstrument
+    class DoNotInstrumentActivity : Activity()
+
+    @DoNotEndAppStart
+    class DoNotEndAppStartActivity : Activity()
+
+    @DoNotInstrument
+    @DoNotEndAppStart
+    class AnnotatedActivity : Activity()
```

Comment:
naming: maybe this could be named more clearly?

```suggestion
    class NoInstrumentationAppStartActivity : Activity()
```